scripts/extensions/videoEditor/src/VideoEditorThumbnail.tsx

```diff
@@ -0,0 +1,378 @@
+import * as React from 'react';
+import {IArticle, ISuperdesk} from 'superdesk-api';
+import {IErrorMessage, ICrop, IVideoProject} from './interfaces';
+import {pick} from 'lodash';
+
+interface IProps {
+    video: HTMLVideoElement;
+    article: IArticle;
+    crop: ICrop;
+    rotate: number;
+    superdesk: ISuperdesk;
+    renditions: IArticle['renditions'];
+    onToggleLoading: (isLoading: boolean, loadingText?: string, type?: 'video' | 'thumbnail') => void;
+    onSave: (renditions: IArticle['renditions']) => void;
+    onError: (err: IErrorMessage) => void;
+    getCropRotate: (crop: ICrop) => ICrop;
+}
+
+interface IState {
+    type: 'capture' | 'upload' | '';
```

Comment:
What does empty string mean? It'd be good to drop it if there's no meaning or rename it to a clearer name or to `null` if it means the value is not set.

Reply:
Let's change it to `null`, we need an empty value to check if preview thumbnail is dirty or not.